packages/pricing/integration-tests/__tests__/services/pricing-module/price-list.spec.ts

```diff
@@ -0,0 +1,372 @@
+import { SqlEntityManager } from "@mikro-orm/postgresql"
+
+import { PriceListService } from "@services"
+
+import { createCurrencies } from "../../../__fixtures__/currency"
+import { createPriceLists } from "../../../__fixtures__/price-list"
+import { initialize } from "../../../../src"
+import { DB_URL, MikroOrmWrapper } from "../../../utils"
+import { PriceListStatus, PriceListType } from "@medusajs/types"
+
+jest.setTimeout(30000)
+
+describe("PriceList Service", () => {
+  let service: PriceListService
+  let testManager: SqlEntityManager
+
+  beforeEach(async () => {
+    await MikroOrmWrapper.setupDatabase()
+    MikroOrmWrapper.forkManager()
+
+    service = await initialize({
+      database: {
+        clientUrl: DB_URL,
+        schema: process.env.MEDUSA_PRICING_DB_SCHEMA,
+      },
+    })
+
+    testManager = MikroOrmWrapper.forkManager()
+    await createCurrencies(testManager)
+    await createPriceLists(testManager)
+  })
+
+  afterEach(async () => {
+    await MikroOrmWrapper.clearDatabase()
+  })
+
+  describe("list", () => {
+    it("should list priceLists", async () => {
+      const priceListResult = await service.listPriceLists()
+
+      expect(priceListResult).toEqual([
+        expect.objectContaining({
+          id: "price-list-1",
+          name: "Default Price List",
+          description: "Default Price List",
+        }),
+
+        expect.objectContaining({
+          id: "price-list-2",
+          name: "Default Price List",
+          description: "Default Price List",
+        }),
+      ])
+    })
+
+    it("should list priceLists by id", async () => {
+      const priceListResult = await service.listPriceLists({
+        id: ["price-list-1"],
+      })
+
+      expect(priceListResult).toEqual([
+        expect.objectContaining({
+          id: "price-list-1",
+          name: "Default Price List",
+          description: "Default Price List",
+        }),
+      ])
+    })
+
+    it("should list priceLists with relations and selects", async () => {
+      const priceListResult = await service.listPriceLists(
+        {
+          id: ["price-list-1"],
+        },
+        {
+          select: ["id", "name", "prices.id"],
+          relations: ["prices"],
+        }
+      )
+
+      const serialized = JSON.parse(JSON.stringify(priceListResult))
+
+      expect(serialized).toEqual([
+        {
+          id: "price-list-1",
+          name: "Default Price List",
+          prices: [
+            {
+              id: "pl-money-amount-USD",
+              price_list_id: "price-list-1", // TODO: this is not correct
```

Comment:
What does this TODO mean? 😄

Reply:
FYI: mikroorm adds the foreign key to the selects when querying relations